solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -814,7 +818,7 @@ private void loadInternal() {
 
     zkSys.initZooKeeper(this, cfg.getCloudConfig());
     if (isZooKeeperAware()) {
-      solrClientCache.setDefaultZKHost(getZkController().getZkServerAddress());
+      getSolrClientCache().setDefaultZKHost(getZkController().getZkServerAddress());
```

Comment:
this line should really be a part of the initialization you put in computeIfAbsent, since that executes lazily.